Source/DebugMod.cs

```diff
@@ -283,7 +296,31 @@ private void SaveSettings()
             instance.Log("Saved");
         }
 
-        private int PlayerDamaged(int damageAmount) => infiniteHP ? 0 : damageAmount;
+        private int PlayerDamaged(int damageAmount)
+        {
+
+            int damage = infiniteHP ? 0 : damageAmount;
+            if (stateOnDeath && (PlayerData.instance.health - damage <= 0))

```

Comment:
Maybe also check for health blue here? Also how does this interact with 1 hp jonis